### PR TITLE
Basic Polymer integration

### PR DESCRIPTION
**Pull request by bshepherdson**
<foam-select> binds a DAO (currently an eval'd expression - ugh!) to a Polymer
array, and then syncs changes between the DAO and the array. (That means both
upstream changes to the DAO and manipulating the array through your Polymer UI).

Check out polymer/index.html for a todo-list demo app.

### CODE REVIEW

**Comment by bshepherdson** [comment]
I updated my email for this repository to be my @google.com one, so hopefully the CLA thing won't whine about it. In any case, I'm a licensed contributor.

File: polymer/foam-select.html [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+        this.dao.put(obj);
+      }.bind(this));
+
+      if (typeof this._ids[obj.id] !== 'undefined') {
+        // Replace that user with this new one.
+        this.feedback_ = true;
+        this.splice('data', this._ids[obj.id], 1, obj);
+        this.feedback_ = false;
+        return;
+      } else {
```
Comment (kgrgreer) [inline polymer/foam-select.html]: No need for else if you've already returned.

File: polymer/foam-select.html [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+    },
+
+    _put: function(obj) {
+      obj = obj.clone();
+      obj.propertyChange.sub(function(sub) {
+        sub.destroy();
+        this.dao.put(obj);
+      }.bind(this));
+
+      if (typeof this._ids[obj.id] !== 'undefined') {
```
Comment (kgrgreer) [inline polymer/foam-select.html]: Match coding standard.

**Comment by bshepherdson** [comment]
PTAL, all addressed.

Two coding style questions:
- My understanding is that `if ( foo ) return;` and similar are allowed, but if it needs to wrap onto two lines, then it needs braces. Correct?
- Loops are spelled `for ( var i = 0; i < foo.length; i++ ) {`, with the semicolons hugging left and spaced right.

**Comment by adamvy-google** [comment]
Loops should be spelled for ( var i = 0 ; i < foo.length ; i++ ) {

spaces on both sides of the semicolon

File: polymer/foam-select.html [hunk 3]
```diff
@@ -0,0 +1,138 @@
+<link rel="import" href="./bower_components/polymer/polymer.html">
+
+<dom-module id="foam-select">
```
Comment (adamvy-google) [inline polymer/foam-select.html:17]: Would be good to have documentation about what this element does as foam-select is not a very descriptive name.

File: polymer/foam-select.html [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+        for ( var j = 0 ; j < splice.addedCount ; j++ ) {
+          addedKeys[splice.object[splice.index + j].id] = true;
+        }
+        // Now everything that's in removed but not in addedKeys needs to be
+        // removed from the DAO.
+        for ( var j = 0 ; j < splice.removed.length ; j++ ) {
+          var obj = splice.removed[i];
+          if ( ! addedKeys[obj.id] ) {
+            delete this._ids[obj.id];
+            window.setTimeout(this._asyncRemove.bind(this, obj), 0);
```
Comment (adamvy-google) [inline polymer/foam-select.html]: Design wise, I would move the setTimeout into the _asyncPut/_asyncRemove functions so you can just call this._asyncPut(obj) here.

File: polymer/foam-select.html [hunk 5]
```diff
@@ -0,0 +1,138 @@
...
+        remove: this._remove.bind(this),
+        reset: this._reset.bind(this),
+        eof: function() {}
+      };
+
+      this.feedback_ = true;
+      this.data = [];
+      this.feedback_ = false;
+
+      this.dao = eval(this.dao);
```
Comment (adamvy-google) [inline polymer/foam-select.html]: It looks like you could change the type of the dao property to string and do this.get(this.dao, window);

https://www.polymer-project.org/1.0/api/#Polymer.Base:method-get

**Comment by adamvy-google** [comment]
LGTM overall, see if my alternative to eval will work.

Feel free to merge this before documenting foam-select

**Pull request merged**